modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/calcite/CalciteTestingApplication.java

```diff
@@ -0,0 +1,351 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.ducktest.tests.calcite;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.ResultSetMetaData;
+import java.sql.Date;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.Period;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Objects;
+import java.math.BigDecimal;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.ignite.internal.ducktest.utils.IgniteAwareApplication;
+
+/**
+ * Tests sql queries for calcite engine
+ */
+public class CalciteTestingApplication extends IgniteAwareApplication {
+
+    /**
+     *
+     */
+    private static class QueryTest {
+        /**
+         * Query.
+         */
+        String qry;
+        /**
+         * Expected results.
+         */
+        List<Object> expectedResults;
+        /**
+         * Expected type.
+         */
+        Class<?> expectedType;
+
+        /**
+         * @param qry             Query.
+         * @param expectedResults Expected results.
+         */
+        QueryTest(String qry, Object... expectedResults) {
+            this.qry = qry;
+            this.expectedResults = Arrays.asList(expectedResults);
+            expectedType = expectedResults[0].getClass();
+        }
+    }
+
+    /**
+     *
+     */
+    @Override public void run(JsonNode jsonNode) throws SQLException {
+        markInitialized();
+
+        try (Connection conn = thinJdbcDataSource.getConnection()) {
+            beforeTest(conn);
```

Comment:
Use newlines to separate semantic units

See https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=57901455#CodingGuidelines-SemanticUnits